storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -396,7 +401,12 @@ private void fixDeltaLog(FileSystem fs, ExternalCommitEntry entry) throws IOExce
                     fixDeltaLogPutCompleteDbEntry(entry);
                     LOG.info("fixed file {}", entry.fileName);
                     return;
-                } catch(Throwable e) {
+                } catch (java.nio.file.FileAlreadyExistsException e) {
+                    LOG.info("file already copied {}:", e.getClass().getSimpleName(), e);
+                    copied = true;
+                    // should we fixDeltaLogPutCompleteDbEntry after this to be safe?
+                    // Don't return so that we mark Db entry as complete
```

Comment:
`Don't return since we still need to mark the DB entry as complete. This will happen when we execute the main try block at the top of the while loop in the next iteration`